src/java/main/org/apache/zookeeper/server/DataTree.java

```diff
@@ -1045,18 +1046,16 @@ private void getCounts(String path, Counts counts) {
         if (node == null) {
             return;
         }
-        Set<String> children = null;
+        String[] children = null;
         int len = 0;
         synchronized (node) {
-            children = node.getChildren();
+            Set<String> childs = node.getChildren();
+            children = childs.toArray(new String[childs.size()]);
```

Comment:
FYI (only): I *guess* some tests broke 'cause `toArray` copies the contents of `childs` to `children` while `node.getChildren()` returns a internal reference (subject to underneath change) to `children`. So, it would need to be something like `children = new HashSet<>(childs);`